File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    row,col = 1,1
    for col in range(1,yellow):
        if yellow%col != 0:
            print("continue")
            continue
        
        row = yellow//col
        
        if ((row+2)+col)*2 == brown:
            break
        
    answer.append(row+2)
    answer.append(col+2)
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.swift
import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    
    for i in 1...Int(sqrt(Double(yellow))) {
        if yellow%i == 0 {
            if ((i+2)+(yellow/i))*2 == brown {
                return [yellow/i+2, i+2]
            }
        }
    }
    return []
}